src/openrct2/actions/RideSetAppearanceAction.hpp

```diff
@@ -22,19 +22,32 @@
 #include "../world/Park.h"
 #include "GameAction.h"
 
+enum class RideSetAppearanceType : uint8_t
```

Comment:
You could move this into the new RideTypes.h files and instead of casting it later to uint8_t you could add DataSerialiserTraits<RideSetAppearanceType>.

Reply:
I don't think this enum should be moved to RideTypes.h it is only used for this game action and for calling this game action. I'll see if I can add it to the DataSerialiserTraits.

Reply:
Its a major pain to add to DataSerialiserTraits without messing around with the headers loads. I think I'll keep the cast.

Reply:
Thats true but thats for what we get for using headers only for game actions, you get to include everything now. But I would still appreciate having it under the data serialiser now that its own type.

Reply:
The other option would be to drop the enum class and make it a normal enum.

Reply:
The reason I would like to see this in the serialiser is because its also shared by logging, if you implement the log member you also get nice output in your log.

Reply:
Well if you can figure out a simple way to achieve that I'm all ears.

Reply:
Fair point, for the time being this is functional so lets leave it for refactoring.